libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts

```diff
@@ -231,6 +233,86 @@ describe('ModalWrapperComponent', () => {
+          .withStaticPageProgress()
+          .build();
+        spectator.detectComponentChanges();
+      });
+
+      afterEach(() => {
+        // Ensure any observers are destroyed:
+        spectator.fixture.destroy();
+      });
+
+      it('should move embedded page progress content to ion toolbar', () => {
+        const ionContentElement = spectator.query('ion-content');
+        const embeddedComponentElement = ionContentElement.firstElementChild;
+
+        const embeddedPageProgress = embeddedComponentElement.querySelector('kirby-page-progress');
+        expect(embeddedPageProgress).toBeNull();
+
+        const pageProgressAsWrapperChild = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgressAsWrapperChild).not.toBeNull();
+      });
+    });
+
+    describe('with dynamic page progress', () => {
+      beforeEach(() => {
+        spectator = modalWrapperTestBuilder
+          .flavor('modal')
+          .withDynamicPageProgress()
+          .build();
+        spectator.detectComponentChanges();
+      });
+
+      afterEach(() => {
+        // Ensure any observers are destroyed:
+        spectator.fixture.destroy();
+      });
+
+      it('should move embedded page progress to wrapper component when rendered', async () => {
+        const pageProgressContent = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgressContent).toBeNull();
+
+        const embeddedComponent = spectator.query(DynamicPageProgressEmbeddedComponent);
+        embeddedComponent.showPageProgress = true;
+        spectator.detectChanges();
+        await TestHelper.waitForResizeObserver();
+
+        const ionContentElement = spectator.query('ion-content');
+        const embeddedComponentElement = ionContentElement.firstElementChild;
+        const embeddedPageProgress = embeddedComponentElement.querySelector('kirby-page-progress');
+        expect(embeddedPageProgress).toBeNull();
+
+        const pageProgressAsWrapperChild = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgressAsWrapperChild).not.toBeNull();
+      });
+
+      it('should remove embedded page progress content from wrapper component when not rendered', async () => {
+        let pageProgress = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgress).toBeNull();
+
+        const embeddedComponent = spectator.query(DynamicPageProgressEmbeddedComponent);
+        embeddedComponent.showPageProgress = true;
+        spectator.detectChanges();
+        await TestHelper.waitForResizeObserver();
+
+        const pageProgressAsWrapperChild = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgressAsWrapperChild).not.toBeNull();
+
+        embeddedComponent.showPageProgress = false;
+        spectator.detectChanges();
+
+        pageProgress = spectator.element.querySelector('kirby-page-progress');
+        expect(pageProgress).toBeNull();
```

Comment:
Please group in Arrange-Act-Assert

Reply:
I've actually tried to give it the same structure as the test on dynamic footer. See [dynamic footer test on master](https://github.com/kirbydesign/designsystem/blob/0cc859e4ab96b43ecb09695c1f39b19baf066e9d/libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.spec.ts#L331).

Reply:
Should we split the test up in two tests which respectively verifies that page progress is rendered or is removed based on an explicitly provided value for showPageProgress?

Reply:
@MartinSkovsen Hmmm not sure, I think this test is fine as is ☝️ (i.e. assert it's not there, assert it gets rendered, assert it gets removed).
But I've left a comment regarding the implementation of this test 👍